pyproject.toml

```diff
@@ -95,6 +95,7 @@ regression = [
 ]
 transformations = [
     "esig>=0.9.7,<0.10",
+    "iisignature>=0.24",
```

Comment:
I think these should have an upper bound as well, that should be current rounded up to 0.1, if I'm not mistaken, @yarnabrina?

Reply:
Yes, it should. Even though it has not been released for over 4 years, we can safely use ">=0.24,<0.25" etc.

However, I've a separate concern. As per https://github.com/bottler/iisignature/issues/20, installing this library needs `numpy` pre-installed. I don't think that's necessarily to be the case for users. Even if I assume it is, as they can manually check the error and address it, how can we use it in our CI?

(Assuming that it's not being tested currently in CI, don't see a new test being added.)

Reply:
what does "pre-installed" mean, here? Do we not have `numpy` as a core dependency?

Reply:
(**DISCLAIMER**: The following is just my understanding, and I'm not at all confident.)

`pip install sktime[all_extras]` download all packages, builds all that's necessary and then installs all of them together. If any of the package installation fails, all gets cancelled. So, even if `numpy` is a core dependency and `iisignature` is soft, they will be installed simultaneously.

Based on the issue, it looks like `numpy` needs to be installed already even before installation of this package starts. Installing together is not an option, as during building step it (`setup.py`) needs `numpy` which will not be present if `pip install sktime[all_extras]` is run on a new blank environment.

Reply:
> Based on the issue, it looks like numpy needs to be installed already even before installation of this package starts. Installing together is not an option

Ouch, really? Then we can't make it a soft dependency in `all_extras`.

We can still add it to the estimator, of course, although that begs the question how we test.

Reply:
hmmm this seems like a real issue, @fkiraly how do you think it's best to proceed? I think it's not acceptable that pip install sktime[all_extras] will return an error.

Reply:
P.S. have added the version upper bound.

Reply:
Yes, there is. You can use pip to install from a remote or local repo with or without extra dependencies.

You can also build and install sktime wheel locally using `build` and `pip`, it's not needed at all to upload to PyPI. But it'll be much simpler if you install from git repository itself, with or without editable option.

Please refer to [development version part of this documentation](https://www.sktime.net/en/stable/installation.html#id6).

Reply:
Thank you for the help! I can confirm that your suspicion is correct, I have the following error when I install it in a new virtual environment: 

Collecting iisignature>=0.24
  Using cached iisignature-0.24.tar.gz (1.3 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/zs/r35ht9w90bzfpdzytk4tfh9m0000gn/T/pip-install-oke9sm8v/iisignature_d008469a531c4a188be319198b5ceca5/setup.py", line 2, in <module>
          import numpy, os, sys
      ModuleNotFoundError: No module named 'numpy'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

@fkiraly @yarnabrina What might be the best way to moving forward?

Reply:
not sure - one option would be to leave it up to the user to ensure the requirements for the backend.

I.e., we can still offer this as an option in the estimator, and not add `iisignature` to the dependency sets but the drawback would be that we can't test the backend.

This might be better than the alternative to not offer it, though. Perhaps doing this with a warning for users that the backend support is experimental.

Not sure, what do you think, @yarnabrina?

Reply:
Also related:

https://github.com/sktime/sktime/pull/5474, and the python 3.10 version bound.

FYI @sz85512678, `SignatureTransformer` currently has a quite restrictive version bound of python <3.10 on it.

Are any of your changes relaxing that requirement? I think it was coming from `esig`. Have you tried to remove the tag and test it on higher python versions?

Reply:
Unfortunately it does still have the python <3.10 version bound, as esig is still used as the front-end. 
A more long term solution could be to replace the signature computation by signax, which is much more recent: https://github.com/anh-tong/signax. It uses JAX, not sure how compatible it is with the sktime eco-system?

Reply:
> It uses JAX, not sure how compatible it is with the sktime eco-system?

`sktime` core is an interface and library layer, so if the dependencies are properly isolated, anything python is compatible. Maybe a nice next project?

Reply:
I see, indeed maybe a nice next project!